src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTests.cs

```diff
@@ -10775,5 +10775,71 @@ record R(int X)
                 "Microsoft.CodeAnalysis: {EmbeddedAttribute}",
                 "System.Runtime.CompilerServices: {NullableAttribute, NullableContextAttribute}");
         }
+
+        [Fact]
+        public void TopLevelStatement_Update()
+        {
+            var source0 =
+    @"using System;
+
+Console.WriteLine(""Hello"");
+";
+            var source1 =
+    @"using System;
+
+Console.WriteLine(""Hello World"");
+";
+            var compilation0 = CreateCompilation(source0, options: TestOptions.DebugExe);
+            var compilation1 = compilation0.WithSource(source1);
+
+            var method0 = compilation0.GetMember<MethodSymbol>("<Program>$.<Main>$");
+
+            // Verify full metadata contains expected rows.
+            var bytes0 = compilation0.EmitToArray();
+            using var md0 = ModuleMetadata.CreateFromImage(bytes0);
+            var reader0 = md0.MetadataReader;
+
+            CheckNames(reader0, reader0.GetTypeDefNames(), "<Module>", "<Program>$");
+            CheckNames(reader0, reader0.GetMethodDefNames(), "<Main>$");
+            CheckNames(reader0, reader0.GetMemberRefNames(), /*CompilationRelaxationsAttribute.*/".ctor", /*RuntimeCompatibilityAttribute.*/".ctor", /*Object.*/".ctor", /*DebuggableAttribute*/".ctor", /*Console.*/"WriteLine");
+
+            var generation0 = EmitBaseline.CreateInitialBaseline(
+                md0,
+                EmptyLocalsProvider);
+            var method1 = compilation1.GetMember<MethodSymbol>("<Program>$.<Main>$");
+
```

Comment:
Nit: I usually define all the versions of the symbols we are interested in together. Feels a little more organized that way.